frontend/src/features/leases/detail/LeasePages/payment/table/columns.tsx

```diff
@@ -0,0 +1,277 @@
+import { Button } from 'components/common/form';
+import { InlineFlexDiv } from 'components/common/styles';
+import TooltipIcon from 'components/common/TooltipIcon';
+import {
+  ColumnWithProps,
+  renderBooleanAsYesNo,
+  renderMoney,
+  renderTypeCode,
+} from 'components/Table';
+import { ILeasePayment } from 'interfaces';
+import { IFormLeaseTerm, ILeaseTerm } from 'interfaces/ILeaseTerm';
+import moment from 'moment';
+import { FaTrash } from 'react-icons/fa';
+import { MdEdit } from 'react-icons/md';
+import { CellProps } from 'react-table';
+import { ISystemConstant } from 'store/slices/systemConstants';
+import styled from 'styled-components';
+import { formatMoney, prettyFormatDate } from 'utils';
+
+function initialOrRenewalTerm({ row: { original, index } }: CellProps<ILeaseTerm, string>) {
```

Comment:
The functions below seem to be doing the lift for the business calculations. It would be better if we have a class/object that would contain the transformations and methods for said transformation while the table columns would only care about how those values are rendered.

Something like: 
ILeaseTerm -> CalculatedLeaseTerm -> PaymentTableRow

Reply:
I agree, would make sense to just update the IFormLeaseTerm to include the calculated rows and do the calculation in the initial conversion. Will try and get this in today, but likely will do it in psp-2020.